lib/IRGen/GenProto.cpp

```diff
@@ -2749,9 +2749,12 @@ static void addAbstractConditionalRequirements(
     auto *proto =
         req.getSecondType()->castTo<ProtocolType>()->getDecl();
     auto ty = req.getFirstType()->getCanonicalType();
-    if (!isa<ArchetypeType>(ty))
+    auto archetype =  dyn_cast<ArchetypeType>(ty);
+    if (!archetype)
       continue;
-    auto conformance = subMap.lookupConformance(ty, proto);
+    auto *genericEnv = archetype->getGenericEnvironment();
+    auto conformance =
+        genericEnv->getForwardingSubstitutionMap().lookupConformance(ty, proto);
```

Comment:
Is this ever not just an abstract conformance? Look at how the forwarding substitution map is constructed...